h/api/models/document.py

```diff
@@ -26,17 +26,18 @@ class Document(Base, mixins.Timestamps):
     def __repr__(self):
         return '<Document %s>' % self.id
 
+    @property
+    def title(self):
+        meta = DocumentMeta.query.filter(DocumentMeta.document_id == self.id,
+                                         DocumentMeta.type == 'title').first()
+        if meta:
+            return meta.value
+
     @classmethod
-    def find_or_create_by_uris(cls, claimant_uri, uris,
```

Comment:
As mentioned in Slack, I'm going to suggest we make a hard break with the way that these kinds of methods are implemented in the rest of the application, and switch to having the convention be that classmethod helpers like these accept the database session as their first argument, rather than using the base class query property. As a result:

- they'll be easier to test
- there won't be any problem with making the session and/or transaction manager entirely request-scoped (rather than thread-local)
- we won't have to do any silliness like [this](https://github.com/hypothesis/h/blob/141e61148fedbbdfb964af884384569569ac5c39/h/db.py#L106) or [this](https://github.com/hypothesis/h/blob/141e61148fedbbdfb964af884384569569ac5c39/scripts/migrate.py#L64).